2023/src/day5/almanac.rs

```diff
@@ -17,6 +17,30 @@ pub struct AlmanacEntry {
   pub kind: AmcEntryKind,
   pub amc_maps: Vec<AlmanacMap>,
 }
+impl AlmanacEntry {
+  pub fn get_dest(&self, src: u64) -> u64 {
+    for amc_map in self.amc_maps.iter() {
+      // check if the src is in the map range
+      let src_end = amc_map.src_start + amc_map.range_len;
+      if src >= amc_map.src_start 
+        && src < src_end
+      {
+        // calculate the offset if it exists
+        let offset: i64 = amc_map.dest_start as i64 - amc_map.src_start as i64;
+        let dest = src.clone() as i64 + offset;
+        return dest as u64;
+      }
+    }
+    /*
+      Return the same src value if no mapping found.
+      Via the prompt:
+        Any source numbers that aren't mapped correspond to
+        the same destination number.
+        So, seed number 10 corresponds to soil number 10.
+    */
+    return src;
```

Comment:
Idiomatically, it would be just `src` here (no `return` or semicolon).

It's kind of a nice Rust-ism because whenever I see `return thing;` I know I'm probably looking at an "unusual" case and not the expected/default outcome.

Reply:
oh, I was trying to do this earlier, and I must've kept the semicolon; I updated this line - thanks!

I will have to get the muscle memory of omitting the `return` and `;`... I like the pattern of `return thing;` being some special case or return-early scenario.